### ISSUE TITLE
Typedecl_separability unbound with Dune

### ISSUE THREAD

**Issue by ghost** [ocaml/ocaml#9294]
Hi all. I am trying to build the compiler using a pre-installed compiler to generate magic number-compatible .merlin map so I can hack on the compiler.

When running `dune build -w @world` it results in:

```
File "typing/typedecl.mli", line 97, characters 20-47:
Error: Unbound module Typedecl_separability
```

Wdyt? Thanks.

**Comment by ghost** [comment]
Possibly related to https://github.com/ocaml/ocaml/pull/2093.

**Comment by trefis** [comment]
Thanks for the report, this is now fixed on trunk by commit d12044e6c206bc4cb27faaefe8953e0b9d3bfee0.

Note: you won't be able to build `@world` with ocaml 4.09 (or older), but there is also the `@libs` alias that will compile the same things but avoid the linking step (which is currently failing).

**Comment by ghost** [comment]
@trefis Thanks!
